app/core/advanced_cache.py

```
@@ -0,0 +1,227 @@
+    
+    def __init__(self, max_size: int = 1000):
+        self._cache: Dict[str, Tuple[Any, Optional[datetime]]] = {}
+        self._max_size = max_size
+        self._lock = threading.Lock()
+        self.stats = CacheStats()
+
+    def get(self, key: str) -> Optional[Any]:
+        with self._lock:
+            if key in self._cache:
+                value, expiry = self._cache[key]
+                if expiry is None or expiry > datetime.now():
```

Ruff (0.8.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)